client/client_test.go

```diff
@@ -58,19 +58,15 @@ func (s *TestSuite) SetUpSuite(t *C) {
 	s.logChan = make(chan proto.LogEntry, 10)
 	s.logger = pct.NewLogger(s.logChan, "ws")
 
-	mock.SendChan = make(chan interface{}, 5)
-	mock.RecvChan = make(chan interface{}, 5)
-	s.server = new(mock.WebsocketServer)
+	s.server = mock.NewWebsocketServer()
 	go s.server.Run(ADDR, ENDPOINT)
 
 	time.Sleep(1 * time.Second)
 
 	links := map[string]string{"agent": URL}
 	s.api = mock.NewAPI("http://localhost", ADDR, "uuid", links)
 
-	mock.SendChanWss = make(chan interface{}, 5)
-	mock.RecvChanWss = make(chan interface{}, 5)
-	s.serverWss = new(mock.WebsocketServerWss)
+	s.serverWss = mock.NewWebsocketServer()
 	go s.serverWss.RunWss(WSSADDR, WSSENDPOINT)
```

Comment:
I've removed `mock.WebsocketServerWss` as the implementation was exactly the same as `mock.WebsocketServer` beside `Run` command. So now we have `WebsocketServer.Run` and `WebsocketServer.RunWss`.